### PR TITLE
GH-44934: [Dev] Remove JIRA remnants

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

We are not using JIRA anymore in the project

### What changes are included in this PR?

Remove JIRA from:
- documentation
- pr and issue templates
- comment bot
- merge script

### Are these changes tested?

Via CI

### Are there any user-facing changes?

No
* GitHub Issue: #44934

### CODE REVIEW

**Comment by raulcd** [comment]
@kou @pitrou this only contains the developer tools updates (merge script, bot comments, pr templates and some minor docs). It does not contain the archery required changes but it's already slightly big. I think it would be better to split the archery changes on a different PR. If you are ok with that I'll create a new issue and would update descriptions accordingly.

**Comment by pitrou** [comment]
Did you forget to update https://github.com/apache/arrow/blob/main/dev/requirements_merge_arrow_pr.txt ?

**Comment by raulcd** [comment]
> Did you forget to update https://github.com/apache/arrow/blob/main/dev/requirements_merge_arrow_pr.txt ?

yes, I forgot :)

File: dev/merge_arrow_pr.py [hunk 1]
```diff
@@ -328,12 +229,13 @@ def format_issue_output(issue_type, issue_id, status,
...
-        url = (
-            f'https://github.com/{ORG_NAME}/{PROJECT_NAME}/issues/{issue_id}'
-        )
+    url_id = issue_id
+    if "GH" in issue_id:
+        url_id = issue_id.replace("GH-", "")
+
+    url = (
+        f'https://github.com/{ORG_NAME}/{PROJECT_NAME}/issues/{url_id}'
+    )
```
Comment (kou) [inline dev/merge_arrow_pr.py:238]: Can we simplify this?

```suggestion
    url = f'https://github.com/{ORG_NAME}/{PROJECT_NAME}/issues/{url_id}'
```

**Review by kou — APPROVED** [review]
+1

Let's work on Archery as a separated task.

**Comment by raulcd** [comment]
I've created this issue to work on the archery changes:
- https://github.com/apache/arrow/issues/44948

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -57,8 +57,8 @@ harder to merge in a large change with a lot of disjoint features.
 GitHub](https://github.com/apache/arrow/issues).
 3. Submit the patch as a GitHub pull request against the main branch. For a
 tutorial, see the GitHub guides on [forking a repo](https://help.github.com/en/articles/fork-a-repo)
-and [sending a pull request](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork). So that your pull request syncs with the JIRA issue, prefix your pull request
-name with the JIRA issue id (ex: [ARROW-767: [C++] Filesystem abstraction](https://github.com/apache/arrow/pull/4225))
+and [sending a pull request](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork). Prefix your pull request
+name with the GitHub issue id (ex: [ARROW-767: [C++] Filesystem abstraction](https://github.com/apache/arrow/pull/4225))
```
Comment (raulcd) [inline .github/CONTRIBUTING.md:61]:
```suggestion
name with the GitHub issue id (ex: [GH-767: [C++] Filesystem abstraction](https://github.com/apache/arrow/pull/4225))
```

**Review by assignUser — APPROVED** [review]

**Comment by assignUser** [comment]
@amoeba can you remove the mentions of jira in the release guide when you update it?

**Comment by amoeba** [comment]
Yep, I'll plan to get that release guide update in for the 19.0.0 release.

**Pull request merged**